### ISSUE TITLE
Wrong convertion LaTEX-code to MathML

### ISSUE THREAD

**Issue by Futyn-Maker** [KaTeX/KaTeX#2593]
**Describe the bug:**
The LATEX-code are converted to MathML wrong, although by visually all OK. The correct MathML is needed to read math formulas by screen-readers.

**To Reproduce:**
1. Try to type the difficult LaTEX-formula which contains, for example, log or trigonometric function. My example is:
```$${\log_{\sqrt{11}} 121}^2$$```
2. see the source MathML-code (not a visual picture of the formula).

**Expected behavior:**
I should get something about this:
```<math display = "block"><msup ><mrow ><msub ><mi>log</mi><msqrt ><mn>11</mn></msqrt></msub><mo>⁡</mo><mn>121</mn></mrow><mn>2</mn></msup></math>```
But I get such MathML:
```<math ><msup ><mrow ><msub ><mo></mo><msqrt ><mn>11</mn></msqrt></msub><mn>121</mn></mrow><mn>2</mn></msup><mrow encoding = "application/x-tex"></mrow></math>```
**Environment (please complete the following information):**
 - KaTeX Version: latest (the problem is relevant even or your demo)
 - Device: laptop, but the problem is relevant for all devices.
 - OS: Windows 10, iOS 14
 - Browser: Chrome, mobile safari
 - Version: Chrome 86, Safari 14

**Additional context:**

The wrong convertion of LaTEX-code breaks an accessibility. I'm sure that the problem is in a KaTEX, because with MathJax these formulas are converted correctly. The problems almost at all formulas with log or trigonometric functions. In some situations the functions are completely disapeared like in the example above. In other situations, for example, there is an excess subscript, and so on.

**Comment by ronkok** [comment]
> the problem is relevant even or your demo
> functions are completely disapeared like in the example above

I cannot reproduce your result in the demo. Your example input is `{\log_{\sqrt{11}} 121}^2`. And you report that a `<mo>` element is empty which should contain `log`. When I examine the output, I find that the same `<mo>` element is 
```
<mo>
   <mi>log</mi><mo>&#x2061;</mo>
<mo>
```

Now one could argue that the KaTeX should not wrap that `<mi>` element with a `<mo>`. But that would be a different discussion. In this discussion, it it incorrect to say that the `<mo>` is empty and the the `log` term is missing.

**Comment by Futyn-Maker** [comment]
@ronkok How do You read output MathML in katex.org demo page?

**Comment by ronkok** [comment]
In Chrome, I right click the element, invoke the `Inspect option`, drill down to the relevant `<math>` element, right click again, invoke the `Copy Outer HTML` option, and copy to a text editor where I can more easily read the code.

**Comment by Futyn-Maker** [comment]
@ronkok OK, I'm sorry for confuse. Actually The problem is that `log` (for example) is wrapped into `<mi></mi>` tags. I'm blind and I use the NVDA screen reader with Access8Math addon and MathPlayer from Design Science, and these programs are interpret such MathML incorrectly. More than, the VoiceOver in iPhone are same behaveour, too. If I remove `<mi></mi` tags, it's OK. Maybe such convertion (with these tags) is really wrong? I very doubt that at minimum 3 programs have the same bug...

**Comment by Futyn-Maker** [comment]
I've edited the heading comment.

**Comment by ronkok** [comment]
@Futyn-Maker Thank you for the comment. I believe that this MathML can be written in more than one way and still meet the W3C spec for MathML. So we want to write the version that works in both visible browsers and in screen readers. I've volunteered quite a lot of time in the last year to cleaning up KaTeX's MathML and your feedback tells me that I have some more work to do.

So here is what is going to happen next. I'm going to go work up a modification of KaTeX. Maybe a couple different versions. I'll put the output up on a test page somewhere and then I'll come back here, give you the URL, and ask you to tell me how they work. When we get something that works properly for all viewers, then I'll turn in a PR to this repository.

Give me a couple of days. I've got some other things on my plate right now.

**Comment by Futyn-Maker** [comment]
OK, thank You for Your interest! I'll wait for Your comment. An accessibility is real important.

**Comment by ronkok** [comment]
Hey, @Futyn-Maker, could you visit the page at https://deploy-preview-2596--katex.netlify.app/docs/support_table.html, and scroll down to the entry for `\operatorname*`? If you can read the `asin` in that entry, then we should have something that works.

**Comment by Futyn-Maker** [comment]
Yes, it seem that all OK. The operator name are spoken correctly. The MathML code which generated the Access8Math addon in NVDA is following (it can bit differ of original KaTEX-code):
`<math ><munder ><mrow ><mi>asin</mi><mo>⁡</mo></mrow><mi>y</mi></munder><mi>x</mi><mrow encoding = "application/x-tex"></mrow></math>`

Andrey
